src/main/java/wtf/g4s8/examples/spaxos/Acceptor.java

```diff
@@ -46,6 +49,8 @@
  */
 public interface Acceptor<T> {
 
+    T getDecision();
```

Comment:
@stepanvalyavskiy two comments:
 - each call to acceptor is simulating network call, so it should be asynchronous message that could be lost. I'd change it to callback, e.g. `getDecision(Consumer<T> callback)`
 -  this function doesn't seems to be a part of Paxos API, I'd create another interface for that and use it for syn protocol

Reply:
- sure, will change to callback.
- `Acceptor` stores a value. Isn't it better to have access method to this value. e.g. `requestValue(Consumer<T> callback)` ?

Reply:
>     * sure, will change to callback.
> 
>     * `Acceptor` stores a value. Isn't it better to have access method to this value. e.g. `requestValue(Consumer<T> callback)` ?

@stepanvalyavskiy we have three different protocols here for updating the value:
 - CAS (TM, RM communication protocol)
 - Paxos (distributing transaction decision)
 - sync to update current transaction state on TM

Of course the implementation (classes) could implement all these protocols (interface) if needed, e.g. resource manager could implement both CAS, Paxos and sync protocols. So I'd split this interfaces to describe the purpose of each of them more precise.